### PR TITLE
Make RPC port assignment more robust

### PR DESCRIPTION
**Pull request by maxcapodi78**

### CODE REVIEW

File: pyaedt/common_rpc.py [hunk 1]
```diff
@@ -46,6 +45,12 @@ def launch_server(port=18000, ansysem_path=None, non_graphical=False):
     >>> launch_server( port=18000)
 
     """
+    port1 = check_port(port)
+    if port == 0:
+        print("Erron. No Available ports.")
```
Comment (MaxJPRey) [inline pyaedt/common_rpc.py:50]:
```suggestion
        print("Error. No Available ports.")
```

File: pyaedt/common_rpc.py [hunk 2]
```diff
@@ -368,6 +373,10 @@ def launch_ironpython_server(aedt_path, non_graphical=False, port=18000, launch_
     >>> my_face_list = client.convert_remote_object(box.faces)
 
     """
+    port1 = check_port(port)
+    if port1 == 0:
+        print("Erron. No Available ports.")
```
Comment (MaxJPRey) [inline pyaedt/common_rpc.py:378]:
```suggestion
        print("Error. No Available ports.")
```

File: pyaedt/rpc/rpyc_services.py [hunk 3]
```diff
@@ -742,7 +770,11 @@ def exposed_start_service(self, hostname, beta_options=None):
         hostname : str
             Hostname.
         """
-        port = random.randint(18001, 20000)
+
+        port = check_port(random.randint(18500, 20000))
+        if port == 0:
+            print("Erron. No Available ports.")
```
Comment (MaxJPRey) [inline pyaedt/rpc/rpyc_services.py:776]:
```suggestion
            print("Error. No Available ports.")
```

File: pyaedt/rpc/rpyc_services.py [hunk 4]
```diff
@@ -30,6 +31,33 @@
...
+    """
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    check = False
+    while not check:
+        try:
+            s.bind((socket.getfqdn(), port))
+            check = True
+        except socket.error as e:
+            port += 1
+            if port > 29999:
```
Comment (MaxJPRey) [inline pyaedt/rpc/rpyc_services.py:55]: @maxcapodi78 Please add a small comment to explain why this limit "29999". I might be missing something obvious.

**Review by MaxJPRey — APPROVED** [review]

File: pyaedt/rpc/rpyc_services.py (hunk 4 above)
Comment (maxcapodi78) [reply pyaedt/rpc/rpyc_services.py:56]: Done

**Pull request merged**